src/pipdeptree/__init__.py

```diff
@@ -41,6 +41,18 @@ def guess_version(pkg_key, default="?"):
     :returns: version
     :rtype: string
     """
+    try:
+        if sys.version_info >= (3, 8):  # pragma: >=3.8 cover
+            import importlib.metadata as importlib_metadata
+        else:  # pragma: <3.8 cover
+            import importlib_metadata
+        return importlib_metadata.version(pkg_key)
+    except ImportError:
+        pass
+    # Avoid AssertionError with setuptools
+    # see https://github.com/tox-dev/pipdeptree/issues/162
+    if pkg_key in {"setuptools"}:
+        return default
```

Comment:
Seems this line is not covered in the test suite, also move the comment into one line

Reply:
So this case triggers only if we have `--local-only` or `--user-only` command line flag, and `setuptools` is not installed in the current venv or the user site packages (possibly available through the system site packages). But there is some package in the venv dependent on `setuptools`, like `pytest` or `ipython`. And I'm not sure how to create such an environment when the tests run.

Reply:
Or well maybe we just need some package dependent on setuptools in the venv, but don't have setuptools installed.

The `--user-only` and `--local-only` thing is only needed to trigger the `AssertionError`.

Reply:
And I just realized I don't need to make a real venv to test it anyway, let me give it a try.

Reply:
Okay I have added a simple test.

I wanted to replicate the AssertionError too, but it was not simple because the test imports `virtualenv`, which imports `distutils` and that applies the `setuptools` hack before `pipdeptree` gets imported. The test that relies on `virtualenv` would start failing if we try to import `pipdeptree` before `virtualenv`.

Reply:
Actually I was being stupid, `virtualenv` imports `disutils` not `setuptools` so that test doesn't really fail if we move the `virtualenv` import below.

So now I have updated the test, so that it would give AssertionError without the changes from this PR.